### DISCUSSION TITLE
How to manage users from multiple customers?

### DISCUSSION THREAD

**Discussion by lusabo** [keycloak/keycloak#22267] [Q&A]
Hello everyone, 

I'm creating a SaaS pilot, this application will be able to serve several customers and each one will have its set of users, I would like to know the best way to do this in Keycloak?

Is it possible to create a route so that each customer, when accessing their url (see example below) the system redirect to the login page where we can identify in a transparent way the customer then should not be not necessary for the user to inform which client he belongs to? 

Example:
http://customerA.myapp.com 
http://customerB.myapp.com

How do I make sure that, when a company signs up for the service, it is already saved inside the Keycloak?

**Comment by CollinEstes** [comment]
One possible approach would be to provision a realm per customer and OIDC clients within each realm that are specific to each customers home page. You would need to understand how to land the right client credentials into you application to know which client to use (or your frontend depending how you redirect for a authentication). 

We run a multi-tenant platform with keycloak and do things this way. We use terraform to manage the provisioning and setup of each customers realm. 

Hope that helps give you some ideas.

**Reply by lusabo** [reply]
Thanks @CollinEstes! 

I'm using the approach of one realm for a tenant, but I'm creating that realm using Keycloak API when the user signs up on the platform. Do you think this kind of approach is bad?
